src/MessageSource.php

```diff
@@ -23,21 +23,19 @@ final class MessageSource implements MessageReaderInterface, MessageWriterInterf
      * @psalm-var array<string, array<string, array<string, array<string, string>>>>
      */
     private array $messages = [];
-    private ConnectionInterface $db;
     private string $sourceMessageTable = '{{%source_message}}';
     private string $messageTable = '{{%message}}';
-
-    private ?CacheInterface $cache;
     private int $cachingDuration = 3600;
 
-    public function __construct(ConnectionInterface $db, ?CacheInterface $cache = null, ?int $cacheDuration = null)
-    {
-        $this->db = $db;
-        $this->cache = $cache;
+    public function __construct(
+        private ConnectionInterface $db,
+        private CacheInterface|null $cache = null,
+        int $cacheDuration = null
```

Comment:
```suggestion
        int $cacheDuration = 3600
```

This way we can remove the property, as far as I can see. Right?